src/backend/joanie/payment/models.py

```diff
@@ -1 +1,430 @@
-# Create your models here.
+"""
+Declare and configure models for the payment part
+"""
+import uuid
+from decimal import Decimal as D
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db import models
+from django.dispatch import receiver
+from django.utils import timezone
+from django.utils.translation import get_language
+from django.utils.translation import gettext_lazy as _
+
+from howard.issuers import InvoiceDocument
+from parler.models import TranslatableModel, TranslatedFields
+from parler.utils.context import switch_language
+
+from joanie.core import enums as core_enums
+from joanie.core.models import Order
+
+from . import enums as payment_enums
+from . import get_payment_backend
+
+User = get_user_model()
+
+
+class Invoice(TranslatableModel):
+    """
+    Invoice model is an accounting element related to an order
+    """
+
+    reference = models.CharField(
+        db_index=True,
+        editable=False,
+        max_length=40,
+        unique=True,
+    )
+
+    created_on = models.DateTimeField(
+        _("created on"),
+        auto_now_add=True,
+        editable=False,
+    )
+
+    updated_on = models.DateTimeField(
+        _("updated on"),
+        auto_now=True,
+        editable=False,
+    )
+
+    order = models.OneToOneField(
+        Order, verbose_name=_("order"), on_delete=models.PROTECT, editable=False
+    )
+
+    amount = models.DecimalField(
+        _(f"amount ({getattr(settings, 'JOANIE_CURRENCY')[1]})"),
+        max_digits=9,
+        decimal_places=2,
+        editable=False,
+    )
+
+    recipient_name = models.CharField(
+        _("invoice recipient"), max_length=40, null=False, blank=False, editable=False
+    )
+
+    recipient_address = models.CharField(
+        _("invoice address"), max_length=255, null=False, blank=False, editable=False
+    )
+
+    translations = TranslatedFields(
```

Comment:
as discuss we would not make this field translatable but the translatable fields in the context would be a mapping : language/text.